tempesta_fw/http_parser.c

```diff
@@ -1781,9 +2101,15 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len)
 
 	/* ----------------    Request Line    ---------------- */
 
+	/* Parser internal initilizers, must be called once per message. */
 	__FSM_STATE(Req_0) {
+		__hbh_parser_init_req(req);
```

Comment:
Why did you create a new parser state for the initialization routine? It seems you can call __hbh_parser_init_*() functions in `tfw_http_msg_alloc()` where base parser is initialized.